frontend/src/store/modules/visitors/actions.js

```diff
@@ -16,4 +33,30 @@ export default {
             context.commit(RESET_BUTTON_FETCHING, data.button);
         }
     },
+    [FETCH_TABLE_DATA]: (context, data) => {
+        if (!data.value) {
+            return;
+        }
+        context.commit(SET_TABLE_DATA_FETCHING);
+        context.commit(GET_ACTIVE_BUTTON)
+            .then(response => {
+                data.button = response.data;
+            });
+
+        periodService.getTimeByPeriod(context.state.selectedPeriod)
+            .then(response => {
+                return factoryVisitorService.create(data.button)
+                    .fetchTableValues(response.startDate, response.endDate, data.groupedParameter);
+            })
+            .then(response => {
+                context.commit(SET_TABLE_DATA, response.data);
+            })
+            .catch(err => {
+                context.commit(RESET_TABLE_DATA_FETCHING);
+                throw err;
+            })
+            .finally(() => {
+                context.commit(RESET_TABLE_DATA_FETCHING);
```

Comment:
probably better to move it in then, as if error will occur the error can be lost and unhandled

Reply:
Moved